_stbt/imgutils.py

```diff
@@ -106,13 +106,34 @@ def _load_image(image, flags):
         absolute_filename = find_user_file(relative_filename)
         if not absolute_filename:
             raise IOError("No such file: %s" % relative_filename)
-        numpy_image = cv2.imread(absolute_filename, flags)
+        numpy_image = imread(absolute_filename, flags)
         if numpy_image is None:
             raise IOError("Failed to load image: %s" %
                           absolute_filename)
         return _ImageFromUser(numpy_image, relative_filename, absolute_filename)
 
 
+def imread(filename, flags=None):
+    if flags is None:
+        cv2_flags = cv2.IMREAD_UNCHANGED
+    else:
+        cv2_flags = flags
+
+    img = cv2.imread(filename, cv2_flags)
+    if img is None:
+        return None
+
+    if img.dtype == numpy.uint16:
+        warn("Image %s has 16 bits per channel. Converting to 8 bits."
+             % filename)
+        img = cv2.convertScaleAbs(img, alpha=1.0 / 256)
```

Comment:
With my implementation:

* Values 0-255 -> 0
* Values 256-511 -> 1
* ...
* Values 65280-65535 -> 255

Because [cv2.convertScaleAbs] rounds down.

[cv2.convertScaleAbs]: https://docs.opencv.org/2.4/modules/core/doc/operations_on_arrays.html#convertscaleabs

Reply:
I see, yep that sounds right.